pkg/volume/util/fsquota/quota_linux.go

```diff
@@ -269,11 +269,17 @@ func clearQuotaOnDir(m mount.Interface, path string) error {
 // don't cache the result because nothing will clean it up.
 // However, do cache the device->applier map; the number of devices
 // is bounded.
-func SupportsQuotas(m mount.Interface, path string) (bool, error) {
+// User namespaces prevent changes to project IDs on the filesystem,
+// ensuring xfs-quota metrics' reliability; hence, userNamespacesEnabled is checked.
+func SupportsQuotas(m mount.Interface, path string, userNamespacesEnabled bool) (bool, error) {
 	if !enabledQuotasForMonitoring() {
 		klog.V(3).Info("SupportsQuotas called, but quotas disabled")
 		return false, nil
 	}
+	if !userNamespacesEnabled {
+		klog.V(3).Info("SupportQuotas called and LocalStorageCapacityIsolationFSQuotaMonitoring enabled, but pod is not in a user namespace")
+		return false, nil
+	}
```

Comment:
The [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1029-ephemeral-storage-quotas) does not describe a hard dependency on user namespaces for GA, only as a beta step. How are you going to graduate it to GA for pods without user namespaces if there is no way how to test it, even as alpha?

Reply:
I don't think it should move to GA for pods without user namespaces, I think we'll always depends on user namespaces for it to be able to be relied upon

Reply:
Please update the KEP then.

Reply:
a section was added here https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1029-ephemeral-storage-quotas#implementation-details-of-using-xfs-quota-in-user-namespace which doesn't imply anything about just beta and not GA. Are you looking for more?

Reply:
Yes, I'd expect it in the `Proposal` chapter. The section you linked discusses only xfs and not ext4

Reply:
@PannagaRao can you submit a KEP update as such?

Reply:
x-posting https://github.com/kubernetes/enhancements/pull/4772